lib/handler/redirect-handler.js

```diff
@@ -228,7 +228,8 @@ function cleanRequestHeaders (headers, removeContent, unknownOrigin) {
       }
     }
   } else if (headers && typeof headers === 'object') {
-    const entries = headers instanceof Headers ? headers.entries() : Object.entries(headers)
+    const iterableHeaderEntries = util.isIterable(headers) ? headers : Object.entries(headers)
+    const entries = headers instanceof Headers ? headers.entries() : iterableHeaderEntries
```

Comment:
```suggestion
    const entries = util.isIterable(headers) ? headers : Object.entries(headers)
```

Reply:
I'm afraid it won't work as native Headers is iterable as well.

<img width="1161" alt="image" src="https://github.com/user-attachments/assets/346dc738-f1b3-4475-b422-2bba60de66dd">

Meanwhile let me know if you want me to decompose that logic in something like that instead of nested ternary.


```
if (headers instanceof Headers) {
  entries = headers.entries();
} else if (util.isIterable(headers)) {
  entries = headers;
} else {
  entries = Object.entries(headers);
}

```

Reply:
Yes, native headers iterate entries.

Reply:
```js
> Array.from(new Headers({ a: 1 }))
[ [ 'a', '1' ] ]
> Object.entries({ a: 1 })
[ [ 'a', 1 ] ]
```

Reply:
Ended up doing that
```
const entries = util.isIterable(headers) ? headers.entries() : Object.entries(headers)
```

instead of

```
const entries = util.isIterable(headers) ? headers : Object.entries(headers)
```

otherwise we can't destructure. That way I believe we are fine :)

Reply:
my opinion:

```js
const entries = typeof headers[Symbol.iterator] === 'function' ? headers[Symbol.iterator]() : Object.entries(headers)
```

Reply:
but that's not correct, just because it's iterable doesn't not neccessarily mean it has `.entries()`. I don't understand what's from with my suggestion?

Reply:
@tsctx it's the same as `const entries = util.isIterable(headers) ? headers : Object.entries(headers)`

Reply:
it's different. `util.isIterable` will catch `util.isIterable` .

Reply:
sorry I mix and match @ronag lgtm